### ISSUE TITLE
./build-support/commons/scripts/build_py_commons.sh fails to publish t.c.zookeeper

### ISSUE THREAD

**Issue by jsirois** [twitter/commons#394]
Looks like so under pants 0.0.35:
```console
$ ./build-support/commons/scripts/build_py_commons.sh
...
INFO] Detected git repository at /home/jsirois/dev/3rdparty/jsirois-twitter-commons on branch jsirois/AURORA-1083

12:51:15 00:00 [main]
               (To run a reporting server: ./pants server)
12:51:15 00:00   [bootstrap]
12:51:15 00:00   [setup]
12:51:15 00:00     [parse]
               Executing tasks in goals: setup-py
12:51:15 00:00   [setup-py]
12:51:15 00:00     [setup-py]
                   Creating setup.py project for PythonLibrary(BuildFileAddress(FilesystemBuildFile(/home/jsirois/dev/3rdparty/jsirois-twitter-commons/src/python/twitter/common/zookeeper/BUILD), zookeeper))Fetching https://raw.github.com/twitter/finagle/master/finagle-thrift/src/main/thrift/tracing.thrift
Fetching https://raw.github.com/twitter/finagle/master/finagle-thrift/src/main/thrift/tracing.thrift
Fetching https://raw.github.com/twitter/finagle/master/finagle-thrift/src/main/thrift/tracing.thrift

FAILURE: No exported target owner found for PythonThriftLibrary(BuildFileAddress(FilesystemBuildFile(/home/jsirois/dev/3rdparty/jsirois-twitter-commons/src/thrift/com/twitter/thrift/BUILD), py-thrift))


12:51:19 00:04   [complete]
               FAILURE
```

Pants introduced stricter semantics in setup-py that ensure each transitively included target without a `provides=setup_py` is uniquely owned and in the same tree.  For python_thrift_libaries in the twitter/commons src/thrift root dependend on by exported targets in src/python, this prevents publishing.  Pants will need to allow for either PythonThriftLibrary targets declaring a `provides=setup_py` or else allowing for synthetic targets to fall outside the exported tree (if uniquely owned).

**Comment by jsirois** [comment]
This blocks #392

**Comment by jsirois** [comment]
Support for provides in PythonThriftLibrary has landed in pantsbuild/pants master and will be available in this Friday's release of 0.0.39

**Comment by jsirois** [comment]
The fix is in @ ba861a184d0c178362284ff21996ef6a3a55cc0d
